src/backend/dx11/src/lib.rs

```diff
@@ -454,20 +449,86 @@ impl hal::Device<Backend> for Device {
         mut size: u64,
         usage: buffer::Usage,
     ) -> Result<UnboundBuffer, buffer::CreationError> {
-        unimplemented!()
+        use buffer::Usage;
+
+        let mut bind = 0;
+
+        if usage.contains(Usage::UNIFORM) { bind |= d3d11::D3D11_BIND_CONSTANT_BUFFER; }
+        if usage.contains(Usage::VERTEX) { bind |= d3d11::D3D11_BIND_VERTEX_BUFFER; }
+        if usage.contains(Usage::INDEX) { bind |= d3d11::D3D11_BIND_INDEX_BUFFER; }
+
+        // TODO: >=11.1
+        if usage.contains(Usage::UNIFORM_TEXEL) ||
+           usage.contains(Usage::STORAGE_TEXEL) { bind |= d3d11::D3D11_BIND_SHADER_RESOURCE; }
+
+        // TODO: how to do buffer copies
+        // if usage.contains(Usage::TRANSFER_SRC) ||
+        //    usage.contains(Usage::TRANSFER_DST) { bind |= d3d11::D3D11_BIND_UNORDERED_ACCESS; }
+
+        Ok(UnboundBuffer {
+            bind,
+            size,
+            requirements: memory::Requirements {
+                size,
+                alignment: 1,
+                type_mask: 0x7,
+            }
+        })
     }
 
     fn get_buffer_requirements(&self, buffer: &UnboundBuffer) -> memory::Requirements {
-        unimplemented!()
+        buffer.requirements
     }
 
     fn bind_buffer_memory(
         &self,
         memory: &Memory,
         offset: u64,
-        buffer: UnboundBuffer,
+        unbound_buffer: UnboundBuffer,
     ) -> Result<Buffer, device::BindError> {
-        unimplemented!()
+        // TODO: offset
+        assert_eq!(0, offset);
+        // TODO: structured buffers
+        assert_eq!(0, unbound_buffer.bind & d3d11::D3D11_BIND_SHADER_RESOURCE);
+        // TODO: change memory to be capable of more than one buffer?
+        // assert_eq!(None, memory.buffer);
+
+        let (usage, cpu) = match (memory.can_download(), memory.can_upload()) {
+            (true,  true)  => (d3d11::D3D11_USAGE_DYNAMIC, d3d11::D3D11_CPU_ACCESS_WRITE),
+            (true,  false) => (d3d11::D3D11_USAGE_STAGING, d3d11::D3D11_CPU_ACCESS_READ | d3d11::D3D11_CPU_ACCESS_WRITE),
+            (false, true)  => (d3d11::D3D11_USAGE_DEFAULT, 0), 
```

Comment:
Could you document the decision for these a bit? I guess default is chosen due to the way data gets flushed via subresource updates during mapping?

Reply:
I mostly took reasoning from GL backend to get something running, but it doesn't look very correct ("can download" = `CPU_VISIBLE | CPU_CACHED`, "upload" = `CPU_VISIBLE`).

I think there are some memory types which dont apply. For example, there doesn't seem to be non-coherent memory. Same for lazily allocated (though this doesnt seem to hard to emulate), and CPU cached memory (?) AFAICT (unless this is implied by `USAGE_STAGING`)..

Would this be more appropriate? (Assuming `CPU_VISIBLE` means read & write access)
```rust
DEVICE_LOCAL               => USAGE_DEFAULT, None
DEVICE_LOCAL | CPU_VISIBLE => USAGE_DYNAMIC, Read/Write
CPU_VISIBLE  | CPU_CACHED  => USAGE_STAGING, Read/Write
```

Not sure where to fit in `USAGE_IMMUTABLE` in to that though